File: C#/Test/problem1.cs
using System;
using System.ComponentModel;
using System.Runtime.ExceptionServices;

public class TestProblem1
{
    class Student
    {
        private int admno;
        private char[] sname = new char[20]; 
        private float eng, math, science; 
        private float ctotal()
        {
            return eng + math + science;
        }

        public void Takedata()
        {
            Console.WriteLine("Enter admission number:"); admno = int.Parse(Console.ReadLine());

            Console.WriteLine("Enter student name:"); 
            string nameString = Console.ReadLine(); 
            for (int i = 0; i < nameString.Length; i++)
            {
                sname[i] = nameString[i];
            }

            Console.WriteLine("Enter English score: "); 
            eng = int.Parse(Console.ReadLine()); 
            Console.WriteLine("Enter Math score: "); 
            math = int.Parse(Console.ReadLine()); 
            Console.WriteLine("Enter Science score:"); 
            science = int.Parse(Console.ReadLine());
        }

    public void Showdata()
    {    
        string nameString = new string(sname); 
        Console.WriteLine("\nAdmission number: " + admno + "\nStudent name: "
            + nameString + "\nEnglish score: " + eng + "\nMath score: " + math
            + "\nScience score:" + science + "\nTotal score: " + ctotal());
    }
}

    public static void Main(string[] args)
    {
        Student s = new Student(); 
        s.Takedata();
        s.Showdata();
    }
}

File: C++/Assignment 1/problem1.cpp
#include <iostream>
using namespace std;

class Student
{
    private:
        int admno;
        char sname[20];
        float eng, math, science;
        float total;

        float ctotal()
        {
            return eng + math + science;
        }
    
    public:
        void Takedata();
        void Showdata();
};

void Student::Takedata()
{
    cout << "Enter admission number ";
    cin >> admno;
    cout << "Enter student name: ";
    cin.ignore(1000, '\n');
    cin.getline(sname, sizeof(sname));
    cout << "Enter English score: "; 
    cin >> eng;
    cout << "Enter Math score: "; 
    cin >> math;
    cout << "Enter Science score: ";
    cin >> science;
}

void Student::Showdata()
{
    cout << "\nAdmission number: " << admno << "\nStudent name: "
            << sname << "\nEnglish score: " << eng << "\nMath score: "
            << math << "\nScience score: " << science << "\nTotal: "
            << ctotal();
}

int main()
{
    Student s1; 
    s1.Takedata(); 
    s1.Showdata(); 
    return 0;
}